core/network/fabric-interop-cc/contracts/interop/manage_assets.go

```diff
@@ -338,3 +401,349 @@ func (s *SmartContract) ClaimAsset(ctx contractapi.TransactionContextInterface,
 
 	return nil
 }
+
+// function to fetch the asset-lock <key, value> from the ledger using contractId
+func fetchAssetLockedUsingContractId(ctx contractapi.TransactionContextInterface, contractId string) (string, AssetLockValue, error) {
+	var assetLockVal = AssetLockValue{}
+	var assetLockKey string = ""
+	assetLockKeyBytes, err := ctx.GetStub().GetState(generateContractIdMapKey(contractId))
+	if err != nil {
+		log.Error(err.Error())
+		return assetLockKey, assetLockVal, err
+	}
+
+	if assetLockKeyBytes == nil {
+		errorMsg := fmt.Sprintf("no contractId %s exists on the ledger", contractId)
+		log.Error(errorMsg)
+		return assetLockKey, assetLockVal, errors.New(errorMsg)
+	}
+
+	assetLockKey = string(assetLockKeyBytes)
+	log.Infof("contractId: %s and assetLockKey: %s\n", contractId, assetLockKey)
+
+	assetLockValBytes, err := ctx.GetStub().GetState(assetLockKey)
+	if err != nil {
+		return assetLockKey, assetLockVal, fmt.Errorf("failed to retrieve from the world state: %+v", err)
+	}
+
+	if assetLockValBytes == nil {
+		errorMsg := fmt.Sprintf("contractId %s is not associated with any currently locked asset", contractId)
+		log.Error(errorMsg)
+		return assetLockKey, assetLockVal, errors.New(errorMsg)
+	}
+
+	err = json.Unmarshal(assetLockValBytes, &assetLockVal)
+	if err != nil {
+		errorMsg := fmt.Sprintf("unmarshal error: %s", err)
+		log.Error(errorMsg)
+		return assetLockKey, assetLockVal, errors.New(errorMsg)
```

Comment:
This 3-line code snippet is repeated about 157 times (roughly 470 LoC) across `manage_assets.go` and `asset_locks.go` with minor variations. I think it clutters the code a bit. Given that all this code does is log the error before creating it, I think we can reduce the three lines to a single call to a helper function. The helper function (`logThenErrorf(...)` or something along those lines) could take a variadic argument of the formatted error string and its parameters, from which it can create an error message, log it and return an associated `error` instance. For places that already have an `error` object (e.g. line 120 `managed_assets.go`), we can have a function that takes the error, logs it, and returns it back.